pkg/release/binaries.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package release
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/release/pkg/binary"
+)
+
+type ArtifactChecker struct {
+	opts *ArtifactCheckerOptions
+	impl artifactCheckerImplementation
+}
+
+type ArtifactCheckerOptions struct {
+	GitRoot  string   // Directory where the repo was cloned
+	Versions []string // Version tags we are checking
+}
+
+func NewArtifactChecker() *ArtifactChecker {
+	return NewArtifactCheckerWithOptions(&ArtifactCheckerOptions{})
+}
+
+func NewArtifactCheckerWithOptions(opts *ArtifactCheckerOptions) *ArtifactChecker {
+	return &ArtifactChecker{
+		opts: opts,
+		impl: &DefaultArtifactCheckerImpl{},
+	}
+}
+
+func (ac *ArtifactChecker) Options() *ArtifactCheckerOptions {
+	return ac.opts
+}
+
+// CheckBinaryTags checks that the binaries produced in the release are
+// correctly tagged with the semver string
+func (ac *ArtifactChecker) CheckBinaryTags() error {
+	for _, tag := range ac.opts.Versions {
+		if err := ac.impl.CheckVersionTags(ac.opts, tag); err != nil {
+			return errors.Wrapf(err, "checking tags in %s binaries", tag)
+		}
+	}
+	return nil
+}
+
+// CheckBinaryArchitectures ensures all the artifacts produced in each
+// release are of the right architecture
+func (ac *ArtifactChecker) CheckBinaryArchitectures() error {
+	for _, tag := range ac.opts.Versions {
+		if err := ac.impl.CheckVersionArch(ac.opts, tag); err != nil {
+			return errors.Wrapf(err, "checking tags in %s binaries", tag)
+		}
+	}
+	return nil
+}
+
+type artifactCheckerImplementation interface {
+	ListReleaseBinaries(opts *ArtifactCheckerOptions, version string) ([]struct{ Path, Platform, Arch string }, error)
+	CheckVersionTags(*ArtifactCheckerOptions, string) error
+	CheckVersionArch(*ArtifactCheckerOptions, string) error
+}
+
+type DefaultArtifactCheckerImpl struct{}
```

Comment:
I think this type can be private 
```suggestion
type defaultArtifactCheckerImpl struct{}
```

Reply:
Done, I've made the type provate